### ISSUE TITLE
add a new option to the connect dialogue and the options dialogue to the current radio buttons, (both controle another computer and allow this computer to be  controled.

### ISSUE THREAD

**Issue by amirmahdifard** [nvda-es/NVDARemote#11]
currently, we are only limited to controling another computer, or allow our computer to be controled: but for my job since i need to controle another computer with my current computer and also controle my current computer with the another computer, so everytime i want to do that i must disconnect, change it to the other radio button, and connect again: and also, i beleave that some people are like to controle their both computers at the same time: and adding this feecher won't be annoyingvfor any people: in the connect dialogue, we should have 3 radio options: 1: controle another computer, 2: allow this computer to be controled, and 3: both, controle and be controled: then, people are completely free to use any option: if people wants to select the both option, the 2 person from both side should select the both radio button: once they did and connected, both people will have speech of the other side computer, they both will be able to press f11 key and controle the other computer while the other computer will also be able to controle this computer, and both sides will also be able to use send controle alt delete option too: anyway, everything will be available for both connected sides: this feecher is really look good to me, please one of the contributers could implement this if possible: thanks: note: i requested the same thing to nvda remote project like long long long a go, but the developer of that project doesn't care about any tikits in that project as i looked at and as i'm awhere, then, @jmdaweb suggested this project to me and i beleave this addon is a good addon and i also beleave @jmdaweb is a good and nice developer, so lets see what will be done about this tikit: note: my tikit in nvda remote addon project has 24 comments and all of the people has liked this suggestion and liked it to be implemented but the developer does not care about it, if you want, i can specify the link to that tikit in that project: please pay attention to this, thank you so much! i'm hoping to this project now.

**Comment by amirmahdifard** [comment]
here's the link to the tikit i have opened in the nvda remote project, you can look at it and see how the developer is unresponcible
https://github.com/NVDARemote/NVDARemote/issues/308
thanks if you can add this ability.

**Comment by amirmahdifard** [comment]
@jmdaweb hi bro, did you see this?

**Comment by jmdaweb** [comment]
Hi,
I'll work on this. It seems possible. Expect a new development version in a few days.
Regards.

**Comment by amirmahdifard** [comment]
thaaaaanks! so no more konstently discommecting and changeing  the radio butonnes and reconnecting, you are greate!

**Comment by amirmahdifard** [comment]
@jmdaweb i have a question: if you add this, do you need to also update nvda remote relay server so it will support this ability, or it has nothing to do with the server and the server will automaticly support this between the clients?

**Comment by amirmahdifard** [comment]
@jmdaweb hi, you told me to wait a fue days so you can add this, but you didn't do: can i ask why? any problem happned?

**Comment by jmdaweb** [comment]
Hi,
Yes and no. I have been busy on other projects these weeks. Maintaining the spanish community is not easy. But I studied the current code for some minutes, and this change requires merging two session classes in a new class which may cause a lot of issues.
After asking the community, another possible solution already implemented in some remote services is switching roles. Would you like a gesture to switch between controlling and controlled without disconnecting?
@davidacm what do you think about this request?
Regards.

**Comment by davidacm** [comment]
Hi,  I just read the suggestion. Although I haven't had this specific need, I think it could be very useful in some scenarios, and I may even end up using it one day.

One case where this would be useful is when we have 2 computers that we use frequently but need to do things on one or the other.
I think switching roles would seem like the most viable option if we want to preserve code integrity.
This change should be allowed from the controlled computer, but that would be a security risk. To avoid it, an extra option could be added when configuring each session, to accept this role change, informing the user about the security risks it represents and that it should only be used on computers we trust completely.

I'll read the comments of the mentioned issue to see if there are any extra suggestions about this, to define what would be the best approach to implement this feature from the user perspective.

I think I could implement the switching roles feature, if that approach were the best option.

For today and tomorrow I don't have access to my development computer, but as soon as I get home I will be able to analyze the possibilities.

**Comment by amirmahdifard** [comment]
@jmdaweb yeah i liked this suggestion but i think implementing this suggestion that i gave is bettar

**Comment by amirmahdifard** [comment]
@jmdaweb ok then, if you can add an option to switch between them with out disconnecting, then please do that: that's at least bettar than nothing: thanks.

**Comment by amirmahdifard** [comment]
@jmdaweb @davidacm hi guys, i'm excited for this suggestion to be done at some point, can you please do something for this at some point? thanks a lot

**Comment by dragonwolfsp** [comment]
I think @davidacm suggestion would be best as @amirmahdifard suggestion could cause a massive loopback conflict where in both computers are trying to control each other. This would result in one sending a command which is than received bye the other and sent back in a loop. This would likely result in both computers locking up completely.
